File: Js/selectionSort.js
// Selection sort written in JavaScript:

let data = [ 10, 3, 6, 9, 8, 5, 2, 1, 4, 7 ];

function selectionSort(arr) {
    for (let i = 0; i < arr.length; i++) {
        let min = i;
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[min]) {
                min = j;
            }
        }
        
        let temp = arr[i];
        arr[i] = arr[min];
        arr[min] = temp;
    }
    return arr;
}

File: Python/selection_sort.py
# Selection sort written in Python:

data = [ 10, 3, 6, 9, 8, 5, 2, 1, 4, 7 ]

def selection_sort(arr):
    for i in range(len(arr)):
        low = i
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[low]:
                low = j
        temp = arr[i]
        arr[i] = arr[low]
        arr[low] = temp
    return arr
